src/format/opamFormat.ml

```diff
@@ -462,10 +462,39 @@ module V = struct
   let package_atom constraints =
     map_option pkgname constraints
 
+  (* These two functions are duplicated from [OpamFormula] but we need to have a
+     it here because of a change on [Block] handling: to have a coherent
+     printing, we must not always discard them *)
+  let rec ands_to_list = function
+    | Empty -> []
+    | And (e,f)
+    | Block (And (e,f)) ->
```

Comment:
Is even this line needed ? Not having it would preserve cases such as `[ "foo" ("bar" & "baz") ]`, while with the line it would be re-printed as `[ "foo" "bar" "baz" ]`. Not sure if it is best, but at least it should ensure that `parse (print x) = x`, which is no longer the case if we don't preserve all blocks ?

The other approach would be to give up on the `Block` constructor, and consequently on preserving user parentheses ; but I think it's best if they remain for now (even if just for preserving camelus 1.2→2.0 rewrite consistency).